[SVG2] Remove `*Glyph*` support

karlcow opened the pull request:
#### dde40db461de1f1ad49740120ff7a198becedbe2
<pre>
[SVG2] Remove `*Glyph*` support
<a href="https://bugs.webkit.org/show_bug.cgi?id=278868">https://bugs.webkit.org/show_bug.cgi?id=278868</a>
<a href="https://rdar.apple.com/135353569">rdar://135353569</a>

Reviewed by NOBODY (OOPS!).

The SVG 2 specification has removed the following elements.
<a href="https://w3c.github.io/svgwg/svg2-draft/eltindex.html">https://w3c.github.io/svgwg/svg2-draft/eltindex.html</a>
* SVGAltGlyphElement
* SVGAltGlyphDefElement
* SVGAltGlyphItemElement
* SVGGlyphElement
* SVGGlyphRefElement
* SVGMissingGlyphElement

For the last 3 years, the SVG elements related to glyph have been
unexposed from the Web.
 <a href="https://commits.webkit.org/255702@main">https://commits.webkit.org/255702@main</a>
<a href="http://wpt.live/svg/historical.html">http://wpt.live/svg/historical.html</a>

No breakage, no bug reports have been signaled since then. It&apos;s probably
time to complete the removal of this code.

* LayoutTests/TestExpectations:
  Rebaselining some tests.
  Deleting the ones which were really about SVG Glyphs
* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSFontFace.cpp:
(WebCore::CSSFontFace::appendSources):
* Source/WebCore/css/CSSFontFaceSource.cpp:
(WebCore::CSSFontFaceSource::load):
(WebCore::CSSFontFaceSource::font):
(WebCore::CSSFontFaceSource::isSVGFontFaceSource const):
* Source/WebCore/css/CSSFontFaceSource.h:
* Source/WebCore/dom/Document.cpp:
(WebCore::isStandardElementName):
* Source/WebCore/loader/cache/CachedSVGFont.cpp:
(WebCore::CachedSVGFont::ensureCustomFontData):
* Source/WebCore/loader/cache/CachedSVGFont.h:
* Source/WebCore/rendering/svg/legacy/SVGResources.cpp:
(WebCore::clipperFilterMaskerTags):
(WebCore::fillAndStrokeTags):
* Source/WebCore/svg/SVGAltGlyphDefElement.cpp: Removed.
* Source/WebCore/svg/SVGAltGlyphDefElement.h: Removed.
* Source/WebCore/svg/SVGAltGlyphElement.cpp: Removed.
* Source/WebCore/svg/SVGAltGlyphElement.h: Removed.
* Source/WebCore/svg/SVGAltGlyphItemElement.cpp: Removed.
* Source/WebCore/svg/SVGAltGlyphItemElement.h: Removed.
* Source/WebCore/svg/SVGEllipseElement.cpp:
* Source/WebCore/svg/SVGFontElement.cpp:
* Source/WebCore/svg/SVGFontFaceElement.cpp:
* Source/WebCore/svg/SVGGElement.cpp:
* Source/WebCore/svg/SVGGlyphElement.cpp: Removed.
* Source/WebCore/svg/SVGGlyphElement.h: Removed.
* Source/WebCore/svg/SVGGlyphRefElement.cpp: Removed.
* Source/WebCore/svg/SVGGlyphRefElement.h: Removed.
* Source/WebCore/svg/SVGMissingGlyphElement.cpp: Removed.
* Source/WebCore/svg/SVGMissingGlyphElement.h: Removed.
* Source/WebCore/svg/SVGTextPositioningElement.cpp:
* Source/WebCore/svg/SVGToOTFFontConversion.cpp: Removed.
* Source/WebCore/svg/SVGToOTFFontConversion.h: Removed.
* Source/WebCore/svg/svgtags.in:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/dde40db461de1f1ad49740120ff7a198becedbe2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/155982 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/29317 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/22499 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/164744 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/109921 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/157853 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/29450 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/29318 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/120759 "Found 19 new test failures: imported/w3c/web-platform-tests/custom-elements/CustomElementRegistry.html imported/w3c/web-platform-tests/custom-elements/registries/define.html imported/w3c/web-platform-tests/custom-elements/registries/valid-custom-element-names.html svg/W3C-SVG-1.1/animate-elem-03-t.svg svg/W3C-SVG-1.1/animate-elem-24-t.svg svg/W3C-SVG-1.1/animate-elem-36-t.svg svg/W3C-SVG-1.1/animate-elem-40-t.svg svg/W3C-SVG-1.1/filters-light-04-f.svg svg/W3C-SVG-1.1/filters-turb-02-f.svg svg/W3C-SVG-1.1/masking-mask-01-b.svg ... (failure)") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/85063 "Passed tests") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/158940 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/22968 "Found 15 new test failures: svg/W3C-SVG-1.1-SE/paths-dom-02-f.svg svg/W3C-SVG-1.1-SE/types-dom-04-b.svg svg/W3C-SVG-1.1/animate-elem-03-t.svg svg/W3C-SVG-1.1/animate-elem-24-t.svg svg/W3C-SVG-1.1/animate-elem-36-t.svg svg/W3C-SVG-1.1/animate-elem-40-t.svg svg/W3C-SVG-1.1/filters-light-04-f.svg svg/W3C-SVG-1.1/filters-turb-02-f.svg svg/W3C-SVG-1.1/masking-mask-01-b.svg svg/W3C-SVG-1.1/pservers-grad-08-b.svg ... (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/140081 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/101448 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/22052 "Found 3 new test failures: imported/w3c/web-platform-tests/custom-elements/CustomElementRegistry.html imported/w3c/web-platform-tests/custom-elements/registries/define.html imported/w3c/web-platform-tests/custom-elements/registries/valid-custom-element-names.html (failure)") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/20188 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/12575 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/131714 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/17913 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/167224 "Built successfully") | | 
| | [✅ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/11398 "Built successfully") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/19525 "Found 4 new test failures: imported/w3c/web-platform-tests/custom-elements/CustomElementRegistry.html imported/w3c/web-platform-tests/custom-elements/registries/define.html imported/w3c/web-platform-tests/custom-elements/registries/valid-custom-element-names.html platform/mac/svg/fonts/svg-font-general.html (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/128880 "Found 20 new test failures: imported/w3c/web-platform-tests/custom-elements/CustomElementRegistry.html imported/w3c/web-platform-tests/custom-elements/registries/define.html imported/w3c/web-platform-tests/custom-elements/registries/valid-custom-element-names.html svg/W3C-SVG-1.1-SE/paths-dom-02-f.svg svg/W3C-SVG-1.1-SE/types-dom-04-b.svg svg/W3C-SVG-1.1/animate-elem-03-t.svg svg/W3C-SVG-1.1/animate-elem-24-t.svg svg/W3C-SVG-1.1/animate-elem-36-t.svg svg/W3C-SVG-1.1/animate-elem-40-t.svg svg/W3C-SVG-1.1/filters-light-04-f.svg ... (failure)") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/28918 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/24240 "Found 4 new test failures: imported/w3c/web-platform-tests/custom-elements/CustomElementRegistry.html imported/w3c/web-platform-tests/custom-elements/registries/define.html imported/w3c/web-platform-tests/custom-elements/registries/valid-custom-element-names.html platform/mac/svg/fonts/svg-font-general.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/129013 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/28840 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/139707 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/86587 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/23837 "Passed tests") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/16505 "Found 4 new test failures: imported/w3c/web-platform-tests/custom-elements/CustomElementRegistry.html imported/w3c/web-platform-tests/custom-elements/registries/define.html imported/w3c/web-platform-tests/custom-elements/registries/valid-custom-element-names.html platform/mac/svg/fonts/svg-font-general.html (failure)") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/28549 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/92505 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/28076 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/28304 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/28200 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

Ahmad-S792 commented:
- SVG Cursor
- SVG TRef
- SVG Locatable

Some more clean-ups to do. :-)

nikolaszimmermann reviewed (approved):
Thanks @karlcow - it's time to get rid of it. rs=me assuming EWS stays green.

Ahmad-S792 commented on Source/WebCore/svg/svgattrs.in:
```diff
@@ -71,7 +71,6 @@ g2
 glyph-name
 glyph-orientation-horizontal
 glyph-orientation-vertical
-glyphRef
```
When I removed `animateColor, it started failing HTML5 Parser tests and have to do:

`animateColor interfaceName=SVGUnknownElement, JSInterfaceName=SVGElement, noConstructor`

Till this not get fixed in tests upstream (Anne did I think), we can do this workaround.

Refer - https://commits.webkit.org/276683@main

Ahmad-S792 replied:
I would personally patch it locally first (rather than fixing tests) and then see which tests are failing genuinely (besides HTML5 Parser and whether they get fixed with this).

karlcow commented:
ok not too worried about the other failures for now, such as SVG1.1 (which is expected given the move from SVG 1.1 to SVG 2)

But these on the other hand are concerning. 

```
imported/w3c/web-platform-tests/custom-elements/CustomElementRegistry.html
imported/w3c/web-platform-tests/custom-elements/registries/define.html
imported/w3c/web-platform-tests/editing/other/delete-in-child-of-html.tentative.html?designMode=on&method=backspace
imported/w3c/web-platform-tests/editing/other/typing-around-link-element-at-collapsed-selection.tentative.html?target=DesignMode&parent=b&child=i
imported/w3c/web-platform-tests/editing/other/typing-around-link-element-at-non-collapsed-selection.tentative.html?target=DesignMode&parent=b
imported/w3c/web-platform-tests/html/syntax/parsing/Document.getElementsByTagName-foreign-01.html
imported/w3c/web-platform-tests/html/syntax/parsing/html5lib_tests11.html?run_type=uri
imported/w3c/web-platform-tests/html/syntax/parsing/html5lib_tests11.html?run_type=write
imported/w3c/web-platform-tests/html/syntax/parsing/html5lib_tests11.html?run_type=write_single
```

Ahmad-S792 commented:
> ok not too worried about the other failures for now, such as SVG1.1 (which is expected given the move from SVG 1.1 to SVG 2)
> 
> But these on the other hand are concerning.
> 
> ```
> imported/w3c/web-platform-tests/custom-elements/CustomElementRegistry.html
> imported/w3c/web-platform-tests/custom-elements/registries/define.html
> imported/w3c/web-platform-tests/editing/other/delete-in-child-of-html.tentative.html?designMode=on&method=backspace
> imported/w3c/web-platform-tests/editing/other/typing-around-link-element-at-collapsed-selection.tentative.html?target=DesignMode&parent=b&child=i
> imported/w3c/web-platform-tests/editing/other/typing-around-link-element-at-non-collapsed-selection.tentative.html?target=DesignMode&parent=b
> imported/w3c/web-platform-tests/html/syntax/parsing/Document.getElementsByTagName-foreign-01.html
> imported/w3c/web-platform-tests/html/syntax/parsing/html5lib_tests11.html?run_type=uri
> imported/w3c/web-platform-tests/html/syntax/parsing/html5lib_tests11.html?run_type=write
> imported/w3c/web-platform-tests/html/syntax/parsing/html5lib_tests11.html?run_type=write_single
> ```

I think once you do changes in `svgtags.in`, it will definitely fix html5lib tests but I am hoping, it might fix others as well.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/c6ad61b63a78188398f7e903130806bcefd0e5ea. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49875

karlcow commented on Source/WebCore/html/parser/HTMLTreeBuilder.cpp:
```diff
@@ -505,7 +505,6 @@ static MemoryCompactLookupOnlyRobinHoodHashMap<AtomString, QualifiedName> create
         SVGNames::diffuseConstantAttr,
         SVGNames::edgeModeAttr,
         SVGNames::filterUnitsAttr,
-        SVGNames::glyphRefAttr,
```
Yes there are things to fix. I’m just trying to rebase to fix tests right now.

annevk replied:
Okay, converted it to a draft. Once you feel it's ready let me know!

karlcow commented:
STATUS: In the process of rebaselining with the current version of main, to have a better overview of the test failures. Don't hold your breath. It will take time.

karlcow commented on Source/WebCore/WebCore.xcodeproj/project.pbxproj:
```diff
@@ -9611,14 +9602,9 @@
-		24D912BB13CA9A9700D21915 /* SVGGlyphRefElement.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = SVGGlyphRefElement.h; sourceTree = "<group>"; };
+		2442BBF81194C9D300D49469 /* HashChangeEvent.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = HashChangeEvent.h; sourceTree = "<group>"; };         2444CEB32ECC4E7F00C09DC6 /* WebTransportConnectionInfo.h */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.h; path = WebTransportConnectionInfo.h; sourceTree = "<group>"; };
+		24F54EAA101FE914000AE741 /* ApplicationCacheHost.cpp */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.cpp.cpp; path = ApplicationCacheHost.cpp; sourceTree = "<group>"; };
+		24F54EAB101FE914000AE741 /* ApplicationCacheHost.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = ApplicationCacheHost.h; sourceTree = "<group>"; };
```
This looks wrong. That must be an error in Rebaselining. Nothing should be added here. probably. The current main has. 

```
		84801955108BAFB300CB2B1F /* FEGaussianBlur.h in Headers */ = {isa = PBXBuildFile; fileRef = 84801953108BAFB300CB2B1F /* FEGaussianBlur.h */; settings = {ATTRIBUTES = (Private, ); }; };
		8482B7461198C35400BFB005 /* HashChangeEvent.h in Headers */ = {isa = PBXBuildFile; fileRef = 2442BBF81194C9D300D49469 /* HashChangeEvent.h */; };
		8482B7521198CB6B00BFB005 /* JSHashChangeEvent.h in Headers */ = {isa = PBXBuildFile; fileRef = 8482B7501198CB6B00BFB005 /* JSHashChangeEvent.h */; };
```

and 

```
		244447FB2ECD1F8E00EED177 /* WebTransportReliabilityMode.h in Headers */ = {isa = PBXBuildFile; fileRef = FAC3C3EF2A996700001E2EB8 /* WebTransportReliabilityMode.h */; settings = {ATTRIBUTES = (Private, ); }; };
		2444CEB42ECC4E7F00C09DC6 /* WebTransportConnectionInfo.h in Headers */ = {isa = PBXBuildFile; fileRef = 2444CEB32ECC4E7F00C09DC6 /* WebTransportConnectionInfo.h */; settings = {ATTRIBUTES = (Private, ); }; };
		24D912B113CA9A1F00D21915 /* SVGAltGlyphDefElement.h in Headers */ = {isa = PBXBuildFile; fileRef = 24D912AE13CA9A1F00D21915 /* SVGAltGlyphDefElement.h */; };
```

But these have been removed.  

```
		24F54EAA101FE914000AE741 /* ApplicationCacheHost.cpp */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.cpp.cpp; path = ApplicationCacheHost.cpp; sourceTree = "<group>"; };
		24F54EAB101FE914000AE741 /* ApplicationCacheHost.h */ = {isa = PBXFileReference; fileEncoding = 4; lastKnownFileType = sourcecode.c.h; path = ApplicationCacheHost.h; sourceTree = "<group>"; };
```

karlcow replied:
ok fixed on my local version.

karlcow commented:
*`LayoutTests/svg`*

```
Expected to fail, but passed: (2)
  svg/W3C-SVG-1.1/paths-data-03-f.svg
  svg/repaint/text-repainting-after-modifying-transform.html


Unexpected flakiness: image-only failures (1)
  svg/text/offset-square-svg-font.html [ Pass ImageOnlyFailure ]


Regressions: Unexpected text-only failures (58)
  svg/W3C-SVG-1.1-SE/paths-dom-02-f.svg [ Failure ]
  svg/W3C-SVG-1.1-SE/types-dom-04-b.svg [ Failure ]
  svg/W3C-SVG-1.1/animate-elem-03-t.svg [ Failure ]
  svg/W3C-SVG-1.1/animate-elem-24-t.svg [ Failure ]
  svg/W3C-SVG-1.1/animate-elem-36-t.svg [ Failure ]
  svg/W3C-SVG-1.1/animate-elem-40-t.svg [ Failure ]
  svg/W3C-SVG-1.1/filters-light-04-f.svg [ Failure ]
  svg/W3C-SVG-1.1/filters-turb-02-f.svg [ Failure ]
  svg/W3C-SVG-1.1/fonts-elem-01-t.svg [ Failure ]
  svg/W3C-SVG-1.1/fonts-elem-02-t.svg [ Failure ]
  svg/W3C-SVG-1.1/fonts-elem-03-b.svg [ Failure ]
  svg/W3C-SVG-1.1/fonts-elem-05-t.svg [ Failure ]
  svg/W3C-SVG-1.1/fonts-elem-06-t.svg [ Failure ]
  svg/W3C-SVG-1.1/fonts-elem-07-b.svg [ Failure ]
  svg/W3C-SVG-1.1/fonts-glyph-02-t.svg [ Failure ]
  svg/W3C-SVG-1.1/fonts-glyph-03-t.svg [ Failure ]
  svg/W3C-SVG-1.1/fonts-kern-01-t.svg [ Failure ]
  svg/W3C-SVG-1.1/masking-mask-01-b.svg [ Failure ]
  svg/W3C-SVG-1.1/pservers-grad-08-b.svg [ Failure ]
  svg/W3C-SVG-1.1/render-elems-06-t.svg [ Failure ]
  svg/W3C-SVG-1.1/render-elems-07-t.svg [ Failure ]
  svg/W3C-SVG-1.1/render-elems-08-t.svg [ Failure ]
  svg/W3C-SVG-1.1/render-groups-01-b.svg [ Failure ]
  svg/W3C-SVG-1.1/render-groups-03-t.svg [ Failure ]
  svg/W3C-SVG-1.1/text-align-08-b.svg [ Failure ]
  svg/W3C-SVG-1.1/text-fonts-03-t.svg [ Failure ]
  svg/W3C-SVG-1.1/text-text-04-t.svg [ Failure ]
  svg/W3C-SVG-1.1/text-text-05-t.svg [ Failure ]
  svg/batik/text/textEffect3.svg [ Failure ]
  svg/batik/text/textPosition2.svg [ Failure ]
  svg/batik/text/xmlSpace.svg [ Failure ]
  svg/custom/glyph-selection-arabic-forms.svg [ Failure ]
  svg/custom/glyph-selection-bidi-mirror.svg [ Failure ]
  svg/custom/glyph-selection-lang-attribute.svg [ Failure ]
  svg/custom/glyph-selection-non-bmp.svg [ Failure ]
  svg/custom/glyph-setting-d-attribute.svg [ Failure ]
  svg/custom/glyph-transformation-with-hkern.svg [ Failure ]
  svg/custom/scrolling-embedded-svg-file-image-repaint-problem.html [ Failure ]
  svg/custom/svg-fonts-in-html.html [ Failure ]
  svg/custom/svg-fonts-no-latin-glyph.html [ Failure ]
  svg/custom/svg-fonts-segmented.xhtml [ Failure ]
  svg/custom/svg-fonts-with-no-element-reference.html [ Failure ]
  svg/custom/svg-fonts-without-missing-glyph.xhtml [ Failure ]
  svg/custom/svg-fonts-word-spacing.html [ Failure ]
  svg/custom/text-linking.svg [ Failure ]
  svg/dom/altGlyph-dom.xhtml [ Failure ]
  svg/foreignObject/text-tref-02-b.svg [ Failure ]
  svg/text/alt-glpyh-on-fallback-font-crash.svg [ Failure ]
  svg/text/kerning.svg [ Failure ]
  svg/text/multichar-glyph.svg [ Failure ]
  svg/text/text-hkern-on-vertical-text.svg [ Failure ]
  svg/text/text-overflow-ellipsis-svgfont-kerning-ligatures.html [ Failure ]
  svg/text/text-overflow-ellipsis-svgfont.html [ Failure ]
  svg/text/text-text-04-t.svg [ Failure ]
  svg/text/text-text-05-t.svg [ Failure ]
  svg/text/text-vkern-on-horizontal-text.svg [ Failure ]
  svg/transforms/text-with-mask-with-svg-transform.svg [ Failure ]
  svg/wicd/test-rightsizing-b.xhtml [ Failure ]
```

webkit-early-warning-system commented:
EWS run on previous version of this PR (hash https://github.com/WebKit/WebKit/commit/36d3388f4547694ee8a828ee9784d1deb55c472b)<details>

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/155225 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/28485 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/21644 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/163985 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/108938 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/ff0c045c-f0fb-4536-8096-23cbae17c8b9) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/157098 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/28625 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/28333 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/120110 "Found 53 new test failures: fast/ruby/ruby-expansion-cjk-2.html fast/ruby/ruby-expansion-cjk-5.html fast/text/text-combine-placement.html imported/w3c/web-platform-tests/custom-elements/CustomElementRegistry.html imported/w3c/web-platform-tests/custom-elements/registries/define.html imported/w3c/web-platform-tests/custom-elements/registries/valid-custom-element-names.html svg/W3C-SVG-1.1-SE/paths-dom-02-f.svg svg/W3C-SVG-1.1-SE/types-dom-04-b.svg svg/W3C-SVG-1.1/animate-elem-03-t.svg svg/W3C-SVG-1.1/animate-elem-24-t.svg ... (failure)") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/84833 "1 flakes 3 failures") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/c3ea7381-c815-43ad-98b6-e9839cdcfc54) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/158184 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/22361 "Found 46 new test failures: fast/css3-text/css3-text-decoration/text-decoration-skip/glyph-inside-underline.html fast/text/text-combine-placement.html svg/W3C-SVG-1.1-SE/paths-dom-02-f.svg svg/W3C-SVG-1.1-SE/types-dom-04-b.svg svg/W3C-SVG-1.1/animate-elem-03-t.svg svg/W3C-SVG-1.1/animate-elem-24-t.svg svg/W3C-SVG-1.1/animate-elem-36-t.svg svg/W3C-SVG-1.1/animate-elem-40-t.svg svg/W3C-SVG-1.1/filters-light-04-f.svg svg/W3C-SVG-1.1/filters-turb-02-f.svg ... (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/139396 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/100805 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/f688cca1-4d4f-4cea-986d-0f6a007576f5) 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/21447 "Found 3 new test failures: imported/w3c/web-platform-tests/custom-elements/CustomElementRegistry.html imported/w3c/web-platform-tests/custom-elements/registries/define.html imported/w3c/web-platform-tests/custom-elements/registries/valid-custom-element-names.html (failure)") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/19502 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/11811 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/131112 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/17232 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/166463 "Built successfully") | | 
| | [✅ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/10619 "Built successfully") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/18841 "Found 10 new test failures: fast/css3-text/css3-text-decoration/text-decoration-skip/decoration-skip-crash-fallback-svg.html fast/ruby/ruby-expansion-cjk-2.html fast/ruby/ruby-expansion-cjk-3.html fast/ruby/ruby-expansion-cjk-5.html fast/ruby/ruby-expansion-cjk.html fast/text/text-combine-placement.html imported/w3c/web-platform-tests/custom-elements/CustomElementRegistry.html imported/w3c/web-platform-tests/custom-elements/registries/define.html imported/w3c/web-platform-tests/custom-elements/registries/valid-custom-element-names.html platform/mac/svg/fonts/svg-font-general.html (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/128215 "Found 52 new test failures: fast/ruby/ruby-expansion-cjk-2.html fast/ruby/ruby-expansion-cjk-5.html fast/text/text-combine-placement.html imported/w3c/web-platform-tests/custom-elements/CustomElementRegistry.html imported/w3c/web-platform-tests/custom-elements/registries/define.html imported/w3c/web-platform-tests/custom-elements/registries/valid-custom-element-names.html svg/W3C-SVG-1.1-SE/paths-dom-02-f.svg svg/W3C-SVG-1.1-SE/types-dom-04-b.svg svg/W3C-SVG-1.1/animate-elem-03-t.svg svg/W3C-SVG-1.1/animate-elem-24-t.svg ... (failure)") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/28029 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/23536 "Found 10 new test failures: fast/css3-text/css3-text-decoration/text-decoration-skip/decoration-skip-crash-fallback-svg.html fast/ruby/ruby-expansion-cjk-2.html fast/ruby/ruby-expansion-cjk-3.html fast/ruby/ruby-expansion-cjk-5.html fast/ruby/ruby-expansion-cjk.html fast/text/text-combine-placement.html imported/w3c/web-platform-tests/custom-elements/CustomElementRegistry.html imported/w3c/web-platform-tests/custom-elements/registries/define.html imported/w3c/web-platform-tests/custom-elements/registries/valid-custom-element-names.html platform/mac/svg/fonts/svg-font-general.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/128352 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/27953 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/139025 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/84662 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/23213 "Passed tests") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/15822 "Found 10 new test failures: fast/css3-text/css3-text-decoration/text-decoration-skip/decoration-skip-crash-fallback-svg.html fast/ruby/ruby-expansion-cjk-2.html fast/ruby/ruby-expansion-cjk-3.html fast/ruby/ruby-expansion-cjk-5.html fast/ruby/ruby-expansion-cjk.html fast/text/text-combine-placement.html imported/w3c/web-platform-tests/custom-elements/CustomElementRegistry.html imported/w3c/web-platform-tests/custom-elements/registries/define.html imported/w3c/web-platform-tests/custom-elements/registries/valid-custom-element-names.html platform/mac/svg/fonts/svg-font-general.html (failure)") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/27646 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/91750 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/27224 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/27454 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/27297 "Built successfully") | | | | </details>

karlcow commented:
Because the confusion came during a discussion outside of this PR.

SVG fonts != Opentype-SVG fonts.
This PR is interested only by SVG fonts.

# OpenType-SVG fonts: NOT affected                              
                                                          
OpenType-SVG support is completely separate from SVG Fonts:                                                                                                                            
- OpenType-SVG uses OTSVGTable.h/.cpp and FontCoreText.cpp (drawOTSVGRun, findOTSVGGlyphs, etc.)
- None of those files are touched by this PR                                                                                                                                           
- CachedSVGFont only handles legacy SVG font files (.svg with `<font>` markup), not OpenType fonts with SVG  tables

karlcow commented:
The 3 tests in WPT which are still using SVG fonts are:

* LayoutTests/imported/w3c/web-platform-tests/svg/fonts/zero-font-size-adjust.html  Inline `<font>` with `<glyph>`
* LayoutTests/imported/w3c/web-platform-tests/svg/fonts/zero-font-size-adjust-no-glyphs-font.html Inline `<font>` + `<font-face>`, no glyphs
* LayoutTests/imported/w3c/web-platform-tests/svg/fonts/zero-font-size-adjust-no-glyphs-zero-height-font.html Same, with `ascent="0"`

They should be SKIPPED and either converted or removed. 
https://wpt.fyi/results/svg/fonts?label=master&label=experimental&aligned

Ahmad-S792 commented:
> The 3 tests in WPT which are still using SVG fonts are:
> 
> * LayoutTests/imported/w3c/web-platform-tests/svg/fonts/zero-font-size-adjust.html  Inline `<font>` with `<glyph>`
> * LayoutTests/imported/w3c/web-platform-tests/svg/fonts/zero-font-size-adjust-no-glyphs-font.html Inline `<font>` + `<font-face>`, no glyphs
> * LayoutTests/imported/w3c/web-platform-tests/svg/fonts/zero-font-size-adjust-no-glyphs-zero-height-font.html Same, with `ascent="0"`
> 
> They should be SKIPPED and either converted or removed. https://wpt.fyi/results/svg/fonts?label=master&label=experimental&aligned

Added here - https://github.com/WebKit/WebKit/pull/42351 by @rwlbuis

karlcow commented:
@annevk is suggesting that instead of deleting the code we put it behind a flag, in case a native app using webkit is using SVG fonts.

karlcow committed:
[SVG2] Remove `*Glyph*` support
https://bugs.webkit.org/show_bug.cgi?id=278868
rdar://135353569

Reviewed by NOBODY (OOPS!).

The SVG 2 specification has removed the following elements.
https://w3c.github.io/svgwg/svg2-draft/eltindex.html
* SVGAltGlyphElement
* SVGAltGlyphDefElement
* SVGAltGlyphItemElement
* SVGGlyphElement
* SVGGlyphRefElement
* SVGMissingGlyphElement

For the last 3 years, the SVG elements related to glyph have been
unexposed from the Web.
 https://commits.webkit.org/255702@main
http://wpt.live/svg/historical.html

No breakage, no bug reports have been signaled since then. It's probably
time to complete the removal of this code.

* LayoutTests/TestExpectations:
  Rebaselining some tests.
  Deleting the ones which were really about SVG Glyphs
* Source/WebCore/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/css/CSSFontFace.cpp:
(WebCore::CSSFontFace::appendSources):
* Source/WebCore/css/CSSFontFaceSource.cpp:
(WebCore::CSSFontFaceSource::load):
(WebCore::CSSFontFaceSource::font):
(WebCore::CSSFontFaceSource::isSVGFontFaceSource const):
* Source/WebCore/css/CSSFontFaceSource.h:
* Source/WebCore/dom/Document.cpp:
(WebCore::isStandardElementName):
* Source/WebCore/loader/cache/CachedSVGFont.cpp:
(WebCore::CachedSVGFont::ensureCustomFontData):
* Source/WebCore/loader/cache/CachedSVGFont.h:
* Source/WebCore/rendering/svg/legacy/SVGResources.cpp:
(WebCore::clipperFilterMaskerTags):
(WebCore::fillAndStrokeTags):
* Source/WebCore/svg/SVGAltGlyphDefElement.cpp: Removed.
* Source/WebCore/svg/SVGAltGlyphDefElement.h: Removed.
* Source/WebCore/svg/SVGAltGlyphElement.cpp: Removed.
* Source/WebCore/svg/SVGAltGlyphElement.h: Removed.
* Source/WebCore/svg/SVGAltGlyphItemElement.cpp: Removed.
* Source/WebCore/svg/SVGAltGlyphItemElement.h: Removed.
* Source/WebCore/svg/SVGEllipseElement.cpp:
* Source/WebCore/svg/SVGFontElement.cpp:
* Source/WebCore/svg/SVGFontFaceElement.cpp:
* Source/WebCore/svg/SVGGElement.cpp:
* Source/WebCore/svg/SVGGlyphElement.cpp: Removed.
* Source/WebCore/svg/SVGGlyphElement.h: Removed.
* Source/WebCore/svg/SVGGlyphRefElement.cpp: Removed.
* Source/WebCore/svg/SVGGlyphRefElement.h: Removed.
* Source/WebCore/svg/SVGMissingGlyphElement.cpp: Removed.
* Source/WebCore/svg/SVGMissingGlyphElement.h: Removed.
* Source/WebCore/svg/SVGTextPositioningElement.cpp:
* Source/WebCore/svg/SVGToOTFFontConversion.cpp: Removed.
* Source/WebCore/svg/SVGToOTFFontConversion.h: Removed.
* Source/WebCore/svg/svgtags.in:

webkit-early-warning-system commented:
EWS run on current version of this PR (hash https://github.com/WebKit/WebKit/commit/dde40db461de1f1ad49740120ff7a198becedbe2)<details>

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/155982 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/29317 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/22499 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/164744 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/109921 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/157853 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/29450 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/29318 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/120759 "Found 19 new test failures: imported/w3c/web-platform-tests/custom-elements/CustomElementRegistry.html imported/w3c/web-platform-tests/custom-elements/registries/define.html imported/w3c/web-platform-tests/custom-elements/registries/valid-custom-element-names.html svg/W3C-SVG-1.1/animate-elem-03-t.svg svg/W3C-SVG-1.1/animate-elem-24-t.svg svg/W3C-SVG-1.1/animate-elem-36-t.svg svg/W3C-SVG-1.1/animate-elem-40-t.svg svg/W3C-SVG-1.1/filters-light-04-f.svg svg/W3C-SVG-1.1/filters-turb-02-f.svg svg/W3C-SVG-1.1/masking-mask-01-b.svg ... (failure)") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/85063 "Passed tests") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/158940 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/22968 "Found 15 new test failures: svg/W3C-SVG-1.1-SE/paths-dom-02-f.svg svg/W3C-SVG-1.1-SE/types-dom-04-b.svg svg/W3C-SVG-1.1/animate-elem-03-t.svg svg/W3C-SVG-1.1/animate-elem-24-t.svg svg/W3C-SVG-1.1/animate-elem-36-t.svg svg/W3C-SVG-1.1/animate-elem-40-t.svg svg/W3C-SVG-1.1/filters-light-04-f.svg svg/W3C-SVG-1.1/filters-turb-02-f.svg svg/W3C-SVG-1.1/masking-mask-01-b.svg svg/W3C-SVG-1.1/pservers-grad-08-b.svg ... (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/140081 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/101448 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/22052 "Found 3 new test failures: imported/w3c/web-platform-tests/custom-elements/CustomElementRegistry.html imported/w3c/web-platform-tests/custom-elements/registries/define.html imported/w3c/web-platform-tests/custom-elements/registries/valid-custom-element-names.html (failure)") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/20188 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/12575 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/131714 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/17913 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/167224 "Built successfully") | | 
| | [✅ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/11398 "Built successfully") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/19525 "Found 4 new test failures: imported/w3c/web-platform-tests/custom-elements/CustomElementRegistry.html imported/w3c/web-platform-tests/custom-elements/registries/define.html imported/w3c/web-platform-tests/custom-elements/registries/valid-custom-element-names.html platform/mac/svg/fonts/svg-font-general.html (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/128880 "Found 20 new test failures: imported/w3c/web-platform-tests/custom-elements/CustomElementRegistry.html imported/w3c/web-platform-tests/custom-elements/registries/define.html imported/w3c/web-platform-tests/custom-elements/registries/valid-custom-element-names.html svg/W3C-SVG-1.1-SE/paths-dom-02-f.svg svg/W3C-SVG-1.1-SE/types-dom-04-b.svg svg/W3C-SVG-1.1/animate-elem-03-t.svg svg/W3C-SVG-1.1/animate-elem-24-t.svg svg/W3C-SVG-1.1/animate-elem-36-t.svg svg/W3C-SVG-1.1/animate-elem-40-t.svg svg/W3C-SVG-1.1/filters-light-04-f.svg ... (failure)") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/28918 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/24240 "Found 4 new test failures: imported/w3c/web-platform-tests/custom-elements/CustomElementRegistry.html imported/w3c/web-platform-tests/custom-elements/registries/define.html imported/w3c/web-platform-tests/custom-elements/registries/valid-custom-element-names.html platform/mac/svg/fonts/svg-font-general.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/129013 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/28840 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/139707 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/86587 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/23837 "Passed tests") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/16505 "Found 4 new test failures: imported/w3c/web-platform-tests/custom-elements/CustomElementRegistry.html imported/w3c/web-platform-tests/custom-elements/registries/define.html imported/w3c/web-platform-tests/custom-elements/registries/valid-custom-element-names.html platform/mac/svg/fonts/svg-font-general.html (failure)") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/28549 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/92505 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/28076 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/28304 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/28200 "Built successfully") | | | | </details>
